### PR TITLE
sqlight: update livecheck

### PR DESCRIPTION
**Pull request by vladimyr**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---

Pass regex as a strategy block param.

### CODE REVIEW

**Comment by bevanjkay** [comment]
@vladimyr I don't believe this change adds readability, especially given that two regexes are used in the livecheck.

**Comment by vladimyr** [comment]
@bevanjkay It enables the reader to quickly figure out the download filename pattern without the need for reading/parsing the strategy block. I'd say that is an improvement compared to the current state of things.

**Comment by miccal** [comment]
Ping @samford for a review.

File: Casks/sqlight.rb [hunk 1]
```diff
@@ -13,10 +13,11 @@
   # needs to be extracted from the download page.
   livecheck do
     url "https://www.aurvan.com/sqlight/"
-    strategy :page_match do |page|
+    regex(/sqlight-v?(\d+(?:[._]\d+)+)-app\.dmg/i)
```
Comment (samford) [inline Casks/sqlight.rb:16]:
```suggestion
    regex(/sqlight[._-]v?(\d+(?:[._]\d+)+)-app\.dmg/i)
```

File: Casks/sqlight.rb [hunk 2]
```diff
@@ -13,10 +13,11 @@
...
+    strategy :page_match do |page, regex|
       js_file = page[%r{src=["']?(/com-aurvan-satva-ui-reactjs-release/static/js/main\.\w+\.chunk.js)["' >]}i, 1]
```
Comment (samford) [inline Casks/sqlight.rb:18]:
```suggestion
      js_file_path = page[%r{src=["']?(/com-aurvan-satva-ui-reactjs-release/static/js/main\.\w+\.chunk\.js)["' >]}i, 1]
      next [] if js_file_path.blank?

```

File: Casks/sqlight.rb [hunk 3]
```diff
@@ -13,10 +13,11 @@
...
       version_page = Homebrew::Livecheck::Strategy.page_content("https://www.aurvan.com/#{js_file}")
-      version_page[:content].scan(/sqlight-v?(\d+(?:[._]\d+)+)-app\.dmg/i).flatten.map { |v| v.tr("_", ".") }
+      version_page[:content].scan(regex).map { |match| match&.first&.gsub("_", ".") }
```
Comment (samford) [inline Casks/sqlight.rb:20]:
```ruby
      js_file_data = Homebrew::Livecheck::Strategy.page_content("https://www.aurvan.com/#{js_file_path}")
      next [] if js_file_data[:content].blank?

      js_file_data[:content].scan(regex).map { |match| match&.first&.gsub("_", ".") }
```

**Review by samford — CHANGES_REQUESTED** [review]
There's a fair amount of history behind this particular question (i.e., using `#regex` and passing it into a `strategy` block vs. defining regexes in a `strategy` block) and I have strong opinions about it but I won't have the time to properly explain it until later today. I plan to open a brew PR to revisit this discussion in earnest and to revert the examples in the livecheck documentation back to its original state (passing `regex` into `strategy` blocks) but that won't happen until after I've worked through some other things on my livecheck to-do list (the last discussion about this was lengthy and time-consuming).

In the interim time, suffice to say that I'm strongly in favor of using `#regex` and passing the value into a `strategy` block in a majority of cases. The main exception is a `strategy` block that uses more than one regex to assemble the version but one regex can't be treated as the primary one over the others (this has only applied to homebrew/cask so far). `sqlight` isn't one of those exceptional cases, as there's only one regex that's used to identify the version and the one in the `livecheck` block is simply to identify the JS file that gets us to the version information.

With that in mind, I've added some suggestions to fix a syntactical issue and to bring this in line with typical patterns for this type of check (with respect to what we do in homebrew/core).

**Pull request merged**